src/context/mod.rs

```diff
@@ -100,6 +100,15 @@ impl AudioContextRegistration {
     pub(crate) fn context(&self) -> &ConcreteBaseAudioContext {
         &self.context
     }
+
+    /// Send a message to the corresponding audio processor of this node
+    ///
+    /// The message will be handled by
+    /// [`AudioProcessor::onmessage`](crate::render::AudioProcessor::onmessage).
+    pub fn post_message(&self, msg: Box<dyn std::any::Any + Send + 'static>) {
```

Comment:
I implemented the *shipping back* pattern for Moire based on `llq`: https://codeberg.org/moire/moire/src/branch/main/crates/ctrl/src/circular_fifo.rs. Not sure if this is suitable here.

The other option would be to hand over handled messages to `basedrop` from the real-time thread.

Reply:
It makes me think we could return the renderers back to the control thread as well when they are removed from the graph? 
Let me know if I'm wrong or badly understand something, but then the memory would be cleared within the control thread and it would also open the door for implementing some pools of renderers?

Reply:
This is tricky. The renderer may have stored/cached AudioRenderQuanta (e.g. DelayRenderer) which are not `Send` so are not allowed to be moved outside of the render thread.

A generic system to decommision AudioProcessors and control messages without touching the global allocator is something worth to explore though.

I will create a separate issue, also taking the recommendations of @uklotzde into account